src/dune_pkg/opam_solver.ml

```diff
@@ -860,6 +880,33 @@ let solve_lock_dir
                       (Package_name.to_string name)
                       (Package_name.to_string dep_name)
                   ]));
+        let reachable =
+          let roots = Package_name.Map.keys local_packages in
+          let pkgs_by_name =
+            Package_name.Map.merge pkgs_by_name local_packages ~f:(fun _name lhs rhs ->
+              match lhs, rhs with
+              | None, None -> assert false
+              | Some _, Some _ -> assert false
```

Comment:
This case should be a code error as it's conceivable that there is a package name in common between local and locked packages, _and_ due to a bug dune failed to detect this case prior to this point.